README.md

```diff
@@ -528,6 +529,24 @@ sslv2, sslv3, tlsv1, sslv23, tlsv11, tlsv12
 The default value for the protocol if not set is tlsv12. The SSLConfig.setEnabled should be called explicitly to enable the SSL.
 The path of the certificate should be correctly provided. 
 
+# AWS Cloud Discovery
+
+Note: You need to provide compile flag -DHZ_BUILD_WITH_SSL when compiling since Aws depends on openssl library.
+
+The C++ client can discover the exiting Hazelcast servers in Amazon AWS environment. The client queries the Amazon AWS environment using the "describe-instances (http://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html)" query of AWS. The client only finds the up and running instances and filters them based on the filter config provided at the ClientAwsConfig configuration.
+ 
+An example configuration:
+```cpp
+clientConfig.getNetworkConfig().getAwsConfig().setEnabled(true).
+            setAccessKey(getenv("AWS_ACCESS_KEY_ID")).setSecretKey(getenv("AWS_SECRET_ACCESS_KEY")).
+            setTagKey("aws-test-tag").setTagValue("aws-tag-value-1").setSecurityGroupName("MySecureGroup").setRegion("us-east-1");
+```
+You need to enable the discovery by calling setEnabled(true). You can set your access key and secret in the config as shown in this example. You can filter the instances by setting which tags they have or by the security group setting. You can set the region for which the instances shall be retrieved from, the default is to use us-east-1.
+ 
```

Comment:
...which the instances will be retrieved.......(shall is usually used with I and we)